src/controllers/company.number.controller.ts

```diff
@@ -0,0 +1,42 @@
+import { Session } from "ch-node-session-handler";
+import { NextFunction, Request, Response } from "express";
+import { ObjectionCompanyProfile } from "../model/objection.company.profile";
+import { OBJECTIONS_CONFIRM_COMPANY } from "../model/page.urls";
+import { getCompanyProfile } from "../services/company.profile.service";
+import { addToObjectionsSession, createObjectionsSession } from "../services/objections.session.service";
+import logger from "../utils/logger";
+
+/**
+ * POST controller for company number screen
+ * @param req
+ * @param res
+ * @param next
+ */
+const route = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+    logger.debug("Attempt to retrieve and show the company details");
+
+    const companyNumber: string = req.body.companyNumber;
+    const session: Session = req.session as Session;
+    logger.info(`Retrieving company profile for company number ${companyNumber}`);
+    const token: string = getValidToken(session, req) as string;
+
+    const company: ObjectionCompanyProfile = await getCompanyProfile(companyNumber, token);
+
+    if (session) {
+        createObjectionsSession(session);
```

Comment:
createObjectionsSession(session); - This is ok for the time being but I think for another story / PR it would be better to create the objections session in middleware, have middleware that calls createObjectionsSession after the sessionMiddleware has run or even as part of the existing session middleware (if objections session doesn't exist). Requires some thought about how we wipe data from the session perhaps, one for another story I guess.

Reply:
Yes this overlaps with further work needed for sessions handling.